eole/predict/beam_search.py

```diff
@@ -306,9 +306,8 @@ def advance(self, log_probs, attn):
 
         if self._stepwise_cov_pen and self._prev_penalty is not None:
             self.topk_log_probs += self._prev_penalty
-            self.topk_log_probs -= self.global_scorer.cov_penalty(
-                self._coverage + attn, self.global_scorer.beta
-            ).view(_B, self.beam_size)
+            cov_penalty = self.global_scorer.cov_penalty(attn, self.global_scorer.beta)
+            self.topk_log_probs -= cov_penalty.view(_B, self.beam_size)
```

Comment:
I don't have clearly understood this part. What do `_stepwise_cov_pen` and `_prev_penalty` mean ?

Reply:
Here is what I gathered quickly looking at the code:

- `_stepwise_cov_pen` is roughly the `stepwise_penalty` condition, which applies penalty at each step
https://github.com/eole-nlp/eole/blob/4a3d0dd3ba9667d0519d846f68b01a1be5fea94e/eole/config/inference.py#L44-L47

- in that context, `_prev_penalty` is just the state at the previous step, so that we accumulate along